src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -90,10 +121,13 @@ internal override ContractTask InitializeAsync(ApplicationEngine engine, Hardfor
                 engine.SnapshotCache.Add(_execFeeFactor, new StorageItem(DefaultExecFeeFactor));
                 engine.SnapshotCache.Add(_storagePrice, new StorageItem(DefaultStoragePrice));
             }
+
             if (hardfork == Hardfork.HF_Echidna)
             {
+                engine.SnapshotCache.Add(_blockGenTime, new StorageItem(engine.ProtocolSettings.MillisecondsPerBlock));
```

Comment:
One thing that I'm not sure about is that this initial value `engine.ProtocolSettings.MillisecondsPerBlock` is not moved to a separate config section. I'm not sure we need it, but I implemented it on the NeoGo side (a separate field in `GenesisConfiguration` section, see the https://github.com/nspcc-dev/neo-go/pull/3835).

It's not critical and we may keep it as is in the C# implementation, but I'd like to hear @roman-khimov's opinion.

Reply:
Configs are implementation-specific, I'd recommend a dedicated "Genesis" section in the "ProtocolConfiguration", but it's just a recommendation.

Reply:
Technically a cn with 16 sec can work with a cn with 15 sec, so maybe we can have a state difference if the node is not well configured

Reply:
It's better to avoid the default value, and take it from settings if is not present, then we can ensure this strange and specific case

Reply:
> Technically a cn with 16 sec can work with a cn with 15 sec

And it looks to me that in this case the "faulty" node should fix its configuration, remove its state and sync from genesis with proper configuration.

Reply:
I mean, avoid write the default value on genesis, and fault if is not present, we already have a method to avoid call this if is not in hf

Reply:
> I mean, avoid write the default value on genesis

But it won't be written to genesis block; it will be written to the block of Echidna activation. And we need the initial value to be a part of the node's state, so that all nodes can rely on this state-based value right after Echidna fork.

Reply:
if we write the default value on genesis we will have state difference for mainnet.

But what I did not understand completely is when we set HF to index 0 on a privatenet, if it will keep getting the default even without consensus calling the method.

Reply:
I missed this line, me previous comments are solved. Sorry for that.

Reply:
My mistake, the way it is here is correct.

Reply:
> > I mean, avoid write the default value on genesis
> 
> But it won't be written to genesis block; it will be written to the block of Echidna activation. And we need the initial value to be a part of the node's state, so that all nodes can rely on this state-based value right after Echidna fork.

How do you will trust on it if during echina my cn write 16 and you 15?

Reply:
> How do you will trust on it if during echina my cn write 16 and you 15?

We're using dBFT, so the dBFT assumption about the number of "correct" nodes in the network works here. At least `M` out of `N` CNs should have valid configurations and hence, valid states.

Reply:
> > > I mean, avoid write the default value on genesis
> > 
> > But it won't be written to genesis block; it will be written to the block of Echidna activation. And we need the initial value to be a part of the node's state, so that all nodes can rely on this state-based value right after Echidna fork.
> 
> How do you will trust on it if during echina my cn write 16 and you 15?

I understand your concern, but it is like Anna highlighted.
2f+1 should have config correct.

In addition, it will lead to state difference,right?

I think that we should follow her advise as well to move this parameter from config.json to genesis.json

Reply:
> > How do you will trust on it if during echina my cn write 16 and you 15?
> 
> We're using dBFT, so the dBFT assumption about the number of "correct" nodes in the network works here. At least `M` out of `N` CNs should have valid configurations and hence, valid states.

But if one of these nodes have a different time per block, can work well and write a different value in the storage in the initialization, because this value was not agree by them, it's in the config of everyone

Reply:
It's a buggy node then. Not much different from invalid standby validator set or invalid HF height. A node with invalid HF height is likely to work, but it's also likely to have a different state. Genesis of HF-dependent values should be synchronized on all nodes.